src/test-run/index.js

```diff
@@ -256,10 +328,37 @@ export default class TestRun extends Session {
             this._fulfillCurrentDriverTask(driverStatus);
         }
 
-        return this.currentDriverTask ? this.currentDriverTask.command : null;
+        return this._getNextTaskCommand();
     }
 
     // Execute command
+    _enqueueHandleDialogCommand (command, callsite) {
+        var dialogType     = command.type.replace(/handle|dialog|-/g, '');
+        var dialog         = new ExpectedDialog(dialogType, command.returnValue);
+        var length         = this.driverTaskQueue.length;
+        var relatedCommand = null;
+        var curCommand     = null;
+
+        // NOTE: we should find a command that triggered the native dialog to add information on how to handle
+        // this dialog. If there's no such a command, we assume that the first page load invoked the native
+        // dialog. We save information about the expected initial dialogs only before the first page load.
+        for (var i = length - 1; i >= 0; i--) {
+            curCommand = this.driverTaskQueue[i].command;
+
+            if (!isHandleDialogCommand(curCommand) && !isObservationCommand(curCommand)) {
+                relatedCommand = curCommand;
+                break;
+            }
+        }
+
+        if (relatedCommand)
+            relatedCommand.expectedDialogs.push(dialog);
+        else if (!this.running)
+            this.initialDialogs.push(dialog);
```

Comment:
so `initialDialog` contains `ExpectedDialog` objects? It should be `initiallyExpectedDialogs` or smthg like that. Same in driver